sway/desktop/output.c

```diff
@@ -273,17 +273,25 @@ static void render_output(struct sway_output *output, struct timespec *when,
 	float clear_color[] = {0.25f, 0.25f, 0.25f, 1.0f};
 	wlr_renderer_clear(renderer, clear_color);
 
-	render_layer(output, &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_BACKGROUND]);
-	render_layer(output, &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_BOTTOM]);
-
 	struct sway_container *workspace = output_get_active_workspace(output);
-	render_container(output, workspace);
 
-	render_unmanaged(output, &root_container.sway_root->xwayland_unmanaged);
+	if (workspace->fullscreen) {
+		wlr_output_set_fullscreen_surface(wlr_output,
+				workspace->fullscreen->surface);
+	} else {
+		wlr_output_set_fullscreen_surface(wlr_output, NULL);
+		render_layer(output,
+				&output->layers[ZWLR_LAYER_SHELL_V1_LAYER_BACKGROUND]);
+		render_layer(output, &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_BOTTOM]);
+
+		render_container(output, workspace);
 
-	// TODO: consider revising this when fullscreen windows are supported
-	render_layer(output, &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_TOP]);
-	render_layer(output, &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY]);
+		render_unmanaged(output, &root_container.sway_root->xwayland_unmanaged);
+
+		render_layer(output, &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_TOP]);
+		render_layer(output,
+				&output->layers[ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY]);
```

Comment:
I think these are still supposed to be rendered, even if there is a fullscreen surface.

Reply:
I disagree, make it an option.

Reply:
We discussed this on IRC and agreed we should add an option for this.

Reply:
We shouldn't give people an option to skip the overlay layer when something is fullscreen, that's just a footgun waiting to happen (it would break lock screens, for example).

Reply:
But we can give them a choice of putting fullscreen views between the shell layer and top layer, or between top and overlay.

Reply:
Fullscreen should certainly hide the bar, right? This is the main difference between fullscreen and maximized.

i3 has [configuration for popups during fullscreen](https://i3wm.org/docs/userguide.html#_popups_during_fullscreen_mode) but that is for transient surfaces, which I don't think we have with xdg-shell.

The overlay should be above the fullscreen surface.

Reply:
Fullscreen should indeed hide the bar. But the bar is in the bottom layer, so it'll work either way.

Reply:
>But the bar is in the bottom layer

This is true for swaybar, but would be false for a GNOME-like/Windows-like/KDE-like bar.

Reply:
I mean, I think we should go between top and overlay by default for that reason. But I'm not gonna sweat it if someone wants to configure it to go elsewhere.